### PR TITLE
Refactor TCA handling to a column-wise approach that enables Tca/Overrides handling

### LINKED ISSUE #2188: Additional report for possible missed out TCA migration addTCAcolumns()
AFAIK we got a new reporting that hints points in the code which cannot fully migrated by rector.

I just found a case in my code which could make use of this:
```
$tmp_showings_columns = [
    'showing' => [
        'exclude' => 1,
        'label' => 'LLL:EXT:offers/Resources/Private/Language/locallang_db.xlf:tx_showings_domain_model_spaceunit.showing',
        'config' => [
            'type' => 'select',
            'foreign_table' => 'tx_showings_domain_model_showing',
            'items' => [
                ['', '']
            ],
            'readOnly' => 1,
        ],
    ],
];
\TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTCAcolumns('tx_offers_domain_model_offer',
    $tmp_showings_columns);
```

The select is not enriched with "selectSingle" as the rector handling this addition ( AddRenderTypeToSelectFieldRector ) checks for full TCA.
As the function does specifically require the columsarray without the wrapping of 'columns' or 'ctrl' which are used to evaluate if the code is actually TCA and should be transformed by rector.

I guess it would be difficult to adapt the rector for this as the conditions are there to prevent transformation of code that is not actually TCA.
So we could filter for the usage of  `addTCAcolumns()` and add that to the reporting

### PR DESCRIPTION
**Pull request by MK-42**
This is a first draft of my thoughts about column-wise tca refactoring.
I chose an easy one to show how the code will look like.

Handling stuff like 'move things into/out of the types section' or other cross-dependencies between other parts of tca and the 'columns' part is still WIP, but please have a look.

I tend to extract some stuff into a base-class such as `AbstractColumnwiseTcaRector`. Or would you prefer to have that in the trait instead @sabbelasichon ?

partially fixes #2188

### CODE REVIEW

File: src/Rector/v8/v6/AddTypeToColumnConfigRector.php [hunk 1]
```diff
@@ -116,4 +77,156 @@ public function getRuleDefinition(): RuleDefinition
...
+    }
+
+    // todo: this should go into a base class
+    private function resolveColumnsDefinition(Node\Expr $columnsDefinition): ?Node
+    {
+        if ($columnsDefinition instanceof Node\Expr\Variable) {
+            // the call to addTcaColumns uses a variable to define the columns to add.
+            // to effectively refactor this argument, we need to find the definition of this variable and refactor that.
+            // as a first-order approximation, we look at the previous statement and hope that the argument is defined there
+            $previousStatement = $columnsDefinition->getAttribute(AttributeKey::PREVIOUS_STATEMENT);
```
Comment (MK-42) [inline src/Rector/v8/v6/AddTypeToColumnConfigRector.php:149]: the following part will probably need some more sophisticated handling. but i would prefer to see such examples in real life code and adapt from here. what do you think?

**Comment by MK-42** [comment]
@sabbelasichon CI failure seems unrelated.

As we discussed, I chose the Array_ node as the root for refactoring now.
As a next step I will adapt the `AddRenderTypeToSelectFieldRector` to this framing to see if it solves our problem at all.

File: src/Rector/Tca/AbstractTcaRector.php [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Expr\ArrayItem;
+use PhpParser\Node\Scalar\String_;
+use Rector\Core\Rector\AbstractRector;
+use Ssch\TYPO3Rector\Helper\TcaHelperTrait;
+
+/**
+ * @changelog
```
Comment (sabbelasichon) [inline src/Rector/Tca/AbstractTcaRector.php:15]: You can exclude some classes to having a @changelog. Add it here: https://github.com/sabbelasichon/typo3-rector/blob/main/utils/phpstan/src/Rules/AddChangelogDocBlockForRectorClass.php#L32

File: tests/Rector/v7/v6/AddRenderTypeToSelectFieldRector/Fixture/add_render_type_in_add_tca_columns.php.inc [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<?php
```
Comment (sabbelasichon) [inline tests/Rector/v7/v6/AddRenderTypeToSelectFieldRector/Fixture/add_render_type_in_add_tca_columns.php.inc:1]: Cool. This is really cool. Looking forward to it.

**Comment by MK-42** [comment]
ready to merge

**Pull request merged**